### ISSUE TITLE
Resizing with R does not always work

### ISSUE THREAD

**Issue by christopho** [solarus-games/solarus-quest-editor#272]
In the map editor, sometimes the R keyboard shortcut to resize entities has no effect.
Workaround: open and close a contextual menu (by right-clicking anywhere) and then it works again.

I only happens sometimes, I don't know the exact conditions to reproduce it.

**Comment by Diarandor** [comment]
Do you know some particular conditions to reproduce it? And for which types of entities does this happen?

**Comment by maxamine** [comment]
it also happens to me sometimes but I don't know how to reproduce the bug.

**Comment by christopho** [comment]
I think it happens for all resizable entities, so most of the time it is with tiles.

I don't know how to reproduce it reliably, that is the main problem.

**Comment by maxamine** [comment]
i'm working on it, trying to finde a way to reproduce it.

**Comment by Diarandor** [comment]
Although I have made very few mapping, I never had this problem, but I never redefined the resizable properties too, so maybe this is related to some particular type of resizable direction property?

**Comment by maxamine** [comment]
i found a way to reproduce the bug, open a map, and another tab (map or whatever).
Now select a tile, resize it using R, after that go on the other tab then back to the map, the tile still selected but R key does no effect.

**Comment by christopho** [comment]
Yes but it works again as soon as you focus the map editor, for example when left-clicking on the selected tile again.

The bug we are trying to address is a more annoying one, because even left clicks everywhere don't fix the situation.

**Comment by maxamine** [comment]
i continue to search then

**Comment by christopho** [comment]
When the bug happens, MapView::start_state_resizing_entities() is not called. Not sure if it is a focus issue.

**Comment by christopho** [comment]
Maybe it is a conflict with the R shortcut of the tileset editor.

**Comment by maxamine** [comment]
If it's a shortcut conflict why it works in most of the time?

**Comment by christopho** [comment]
No idea. I am just logging the progress here, what I found out and some potential leads.

**Comment by maxamine** [comment]
ok i see

**Comment by christopho** [comment]
Reopening, the bug is still here even if when tileset view does not use any R shortcut.

More info: when you have the bug on a map, it can still work in other maps (in other tabs). So there must be something wrong in the state of the map editor.
